src/app/modules/shared-components/components/path-suggestion/path-suggestion.component.ts

```diff
@@ -28,14 +28,19 @@ export class PathSuggestionComponent implements OnDestroy, OnChanges {
 
   state$ = this.itemId$.pipe(
     switchMap(itemId => this.getBreadcrumbsFromRootsService.get(itemId).pipe(
-      map(group =>
-        (group.length > 0 ? group.map(breadcrumbs =>
-          breadcrumbs.map((item, index) => ({
-            ...item,
-            url: getItemRouteUrl(item, breadcrumbs.slice(0, index)),
-          }))
-        ).filter(group => group.length > 0) : undefined)
-      ),
+      map(group => {
+        const allBreadcrumbs = group.reduce((prev, curr) => [ ...prev, ...curr ], []);
+        const root = allBreadcrumbs.length === 1 ? allBreadcrumbs[0] : undefined;
```

Comment:
I don't get the point of the reduce. Why cannot you just write `root = group.length === 1 && group[0].length === 1` ... it could even be "computed" in the template directly.

Reply:
Fixed